src/components/modules/toolbar/inline.ts

```diff
@@ -362,20 +378,76 @@ export default class InlineToolbar extends Module {
     }
 
     const toolSettings = this.Editor.Tools.getToolSettings(currentBlock.name);
+    const inlineToolbarSettingsForTool = toolSettings && toolSettings[this.Editor.Tools.USER_SETTINGS.ENABLED_INLINE_TOOLS];
 
-    return toolSettings && toolSettings[this.Editor.Tools.USER_SETTINGS.ENABLED_INLINE_TOOLS];
+    /**
+     * Condition 1 : Returns true, if the inlineToolbar property of the block is an array or true;
+     * false otherwise.
+     */
+    const isInlineToolbarSettingsForToolProvided = inlineToolbarSettingsForTool === true || Array.isArray(inlineToolbarSettingsForTool);
+
+    /**
+     * Condition 2 : Returns true, if the inlineToolbar property of the block is not set,
+     * but the default inlineToolbar property is either an array or true.
+     */
+    const isDefaultInlineToolbarProvided = inlineToolbarSettingsForTool === undefined && this.config.inlineToolbar;
+
+    /**
+     * If one of the above condition is true, then the inlineToolbar will be displayed.
+     */
+    const showInlineToolbar = isInlineToolbarSettingsForToolProvided || isDefaultInlineToolbarProvided;
+
+    /**
+     * condition 2 could return an array, so casting it to Boolean.
+     */
+    return Boolean(showInlineToolbar);
```

Comment:
```suggestion
    return showInlineToolbar;
```

Reply:
but be sure that code above is safe

Reply:
No its actually not, one of the property returns the inlineToolbar array itself, that's why I used Boolean(). WHat sould I do?